test/controllers/TransferSummaryControllerSpec.scala

```diff
@@ -0,0 +1,132 @@
+        .willReturn(okJson(dataString)))
+
+
+      val transferSummaryPage = controller.transferSummary(consignmentId)
+        .apply(FakeRequest(GET, s"/consignment/$consignmentId/transfer-summary").withCSRFToken)
+
+
+      playStatus(transferSummaryPage) mustBe OK
+      contentType(transferSummaryPage) mustBe Some("text/html")
+      contentAsString(transferSummaryPage) must include("transferSummary.header")
+      contentAsString(transferSummaryPage) must include("transferSummary.seriesReference")
+      contentAsString(transferSummaryPage) must include("transferSummary.transferringBody")
+      contentAsString(transferSummaryPage) must include("transferSummary.filesUploadedForTransfer")
+      contentAsString(transferSummaryPage) must include("transferSummary.openRecords")
+      contentAsString(transferSummaryPage) must include("transferSummary.transferLegalOwnership")
+    }
+
+    "return a redirect to the auth server with an unauthenticated user" in {
+      val consignmentId = UUID.fromString("c2efd3e6-6664-4582-8c28-dcf891f60e68")
+      val controller = new TransferSummaryController(getUnauthorisedSecurityComponents, new GraphQLConfiguration(app.configuration),
+        getValidKeycloakConfiguration, langs)
+      val transferSummaryPage = controller.transferSummary(consignmentId).apply(FakeRequest(GET, "/consignment/123/transfer-summary"))
+
+      redirectLocation(transferSummaryPage).get must startWith("/auth/realms/tdr/protocol/openid-connect/auth")
+      playStatus(transferSummaryPage) mustBe FOUND
+    }
+
+    "return a not found error page if an existing consignment does not have a transfer summary yet" in {
+      val client = new GraphQLConfiguration(app.configuration).getClient[gc.Data, gc.Variables]()
+      val data: client.GraphqlData = client.GraphqlData(Some(gc.Data(None)), List())
+      val dataString: String = data.asJson.printWith(Printer(dropNullValues = false, ""))
+      wiremockServer.stubFor(post(urlEqualTo("/graphql"))
+        .willReturn(okJson(dataString)))
+
+      val consignmentId = UUID.randomUUID()
+      val controller = new TransferSummaryController(getAuthorisedSecurityComponents, new GraphQLConfiguration(app.configuration),
+        getValidKeycloakConfiguration, langs)
+
+      val transferSummaryPage = controller.transferSummary(consignmentId)
+        .apply(FakeRequest(GET, s"/consignment/$consignmentId/transfer-summary").withCSRFToken)
+
+      playStatus(transferSummaryPage) mustBe NOT_FOUND
+      contentAsString(transferSummaryPage) must include("404")
+    }
+
+    "throws an authorisation exception when the user does not have permission to see a consignment's transfer summary" in {
+      val client = new GraphQLConfiguration(app.configuration).getClient[gc.Data, gc.Variables]()
+      val data: client.GraphqlData = client.GraphqlData(Some(gc.Data(None)), List(GraphQLClient.Error("Error", Nil, Nil, Some(Extensions(Some("NOT_AUTHORISED"))))))
```

Comment:
Probably want to put a line break into line 101 as its along line as it is.

Reply:
Ah, yes. I saved it to a variable